src/formatDate.js

```diff
@@ -50,7 +50,43 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = toFormat[3];
+  const dateToArray = date.split(fromFormat[3]);
+  const fromDates = {};
+  const toDates = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    fromDates[fromFormat[i]] = dateToArray[i];
+    toDates[toFormat[i]] = 0;
+  }
+
+  let keyYearFrom = '';
+
+  for (const key in fromDates) {
+    if (key.includes('YY')) {
+      keyYearFrom = key;
+    }
+  }
+
+  let keyYearTo = '';
+
+  for (const key in toDates) {
+    if (key.includes('YY')) {
+      keyYearTo = key;
+    }
+  }
+
+  if (keyYearFrom.length > keyYearTo.length) {
+    fromDates[keyYearTo] = fromDates[keyYearFrom].toString().slice(2);
+  }
+
+  fromDates[keyYearTo] = keyYearFrom.length < keyYearTo.length && fromDates[keyYearFrom] < 30 ? `20${fromDates[keyYearFrom]}`
+    : fromDates[keyYearTo];
+
+  fromDates[keyYearTo] = keyYearFrom.length < keyYearTo.length && fromDates[keyYearFrom] >= 30 ? `19${fromDates[keyYearFrom]}`
+    : fromDates[keyYearTo];
```

Comment:
i mean we can make something like 
```
(fromDates[keyYearFrom] < 30 ? '20' : '19') + fromDates[keyYearFrom]
```

Reply:
When I use such a ternary operator, the tests fail because the second condition is not checked.

<img width="1680" alt="Снимок экрана 2023-08-31 в 21 01 09" src="https://github.com/mate-academy/js_format_date_advanced/assets/118368468/6c74b882-71a6-4c6b-ab10-16a3de5ea37c">